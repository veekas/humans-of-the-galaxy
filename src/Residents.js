import React, { Component } from 'react';

import { getPlanetResidents } from './api';

const ResidentList = ({ residents }) => (
  residents.length
    ? residents.map(resident => <li key={resident.name}>{resident.name}</li>)
    : <div>This planet has no residents.</div>
);

class Residents extends Component {
  constructor(props) {
    super(props);
    this.state = {
      residents: null,
    };
  }

  componentDidMount = () => {
    const { residentUrls } = this.props;
    getPlanetResidents(residentUrls)
      .then(residents => this.setState({ residents }));
  }

  componentWillUnmount = () => {
    console.log(this.statePromises)
  }

  render() {
    const { residents } = this.state;

    return (
      <ul>
        {
          residents
            ? <ResidentList residents={residents} />
            : <div>Loading...</div>
        }
      </ul>
    );
  }
}

export default Residents;
