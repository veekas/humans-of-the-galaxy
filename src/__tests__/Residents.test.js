import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Residents from '../Residents';

enzyme.configure({ adapter: new Adapter() });

describe('<Residents />', () => {
  it('renders successfully', () => {
    const props = { residentUrls: ['https://swapi.co/api/people/1'] };
    const component = shallow(<Residents {...props} />);
    expect(component).toMatchSnapshot();
  });
});
