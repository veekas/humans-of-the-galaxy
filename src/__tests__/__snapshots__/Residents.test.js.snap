// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Residents /> renders successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Residents
    residentUrls={
        Array [
            "https://swapi.co/api/people/1",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.li>
        <styled.img
                alt="Loading..."
                src="loading.svg"
        />
</styled.li>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.img
          alt="Loading..."
          src="loading.svg"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Loading...",
          "src": "loading.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.li>
          <styled.img
                    alt="Loading..."
                    src="loading.svg"
          />
</styled.li>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.img
            alt="Loading..."
            src="loading.svg"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Loading...",
            "src": "loading.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
